src/predict.rs

```diff
@@ -74,7 +74,7 @@ impl Dim for Block4x4 {
 }
 
 pub fn pred_dc_trait<D: Dim>(output: &mut [u16], stride: usize, above: &[u16], left: &[u16]) {
-    let edges = left.iter().chain(above.iter());
+    let edges = left[..D::W].iter().chain(above[..D::H].iter());
```

Comment:
If `above` is the row of pixels above the block, it should have size `D::W`.
If `left` is the column of pixels to the left of the block, it should have size `D::H`.